mlflow/models/signature.py

```diff
@@ -338,13 +338,19 @@ def _infer_signature_from_input_example(
         `wrapped_model`.
     """
     try:
-        input_example = _Example(input_example)
-        # Copy the input example so that it is not mutated by predict()
-        input_ex = deepcopy(input_example.inference_data)
-        params = input_example.inference_params
-        input_schema = _infer_schema(input_ex)
+        if _contains_params(input_example):
+            input_example, params = input_example
+        else:
+            params = None
+        if isinstance(input_example, list) and all(isinstance(x, str) for x in input_example):
+            input_schema = _infer_schema(input_example)
+        else:
+            input_example = _Example(input_example)
```

Comment:
So we're not going to make a breaking change for `_Example` class for now? Do we plan to do so in the future?

Reply:
We're going to add support for array & object in model signature in the future, then this doesn't matter.